### ISSUE TITLE
Explain that `cabal repl --with-ghc=doctest-parallel` can't exist

### ISSUE THREAD

**Issue by ysangkok** [martijnbastiaan/doctest-parallel#36]
In the "Future of this project" section, it is said that

> It would be lovely if we could get rid of the needs for write-ghc-environment-files: always

Doctest-0.20 allows for a new repl invocation mechanism, mentioned in [their changelog](https://hackage.haskell.org/package/doctest-0.20.0/changelog). For Servant, it seems to [enable getting rid of the mentioned flag](https://github.com/haskell-servant/servant/pull/1554/files).

It seems that this trick can't work for doctest-parallel, since there is no executable to reference. Is that correct? Would you accept a PR explaining this?

**Comment by martijnbastiaan** [comment]
Yes, that's correct. Using `cabal repl` would get us back to running against the interpreted code, which is something `doctest-parallel` is trying to avoid. 

> Would you accept a PR explaining this?

Absolutely!

As a side-note; [cabal-docspec](https://github.com/phadej/cabal-extras/tree/master/cabal-docspec) parses files in `dist-newstyle` to find libraries and package dbs. This would be a way to get rid of environment files. I opted not to when I started working on `doctest-parallel`, because I felt it was too much effort for something that other build tools (stack/nix) handle without any custom logic.
